client/my-sites/checkout/utils.ts

```diff
@@ -47,7 +47,7 @@ export function getProductSlugFromContext(
 	}
 
 	if ( isGiftPurchase ) {
-		return domainOrProduct;
+		return domainOrProduct || product;
```

Comment:
This seems weird. Is it something that could use a comment in code explaining why this is necessary?

Reply:
Looking through routes, the only `/gift/` route I see uses `:product` I think I'm misunderstanding this or missing a `:domainOrProduct` path.

Reply:
Yeah, I agree actually. I'll revisit this.